canonicalwebteam/blog/wordpress_api.py

```diff
@@ -47,24 +49,27 @@ def build_get_articles_url(
 
     :returns: URL to Wordpress api
     """
-    url = (
-        f"{API_URL}/posts?per_page={per_page}"
-        f"&tags={','.join(str(id) for id in tags)}"
-        f"&page={page}"
-        f"&group={','.join(str(id) for id in groups)}"
-        f"&tags_exclude={','.join(str(id) for id in tags_exclude)}"
-        f"&categories={','.join(str(id) for id in categories)}"
-        f"&exclude={','.join(str(id) for id in exclude)}"
-        f"&author={author}&_embed"
-    )
-    if sticky != "":
-        url = url + f"&sticky={sticky}"
-    if before != "":
-        url = url + f"&before={before}"
-    if after != "":
-        url = url + f"&after={after}"
-
-    return url
+    params_object = [
+        param
+        for param in [
+            ("tags", ",".join(str(id) for id in tags)),
```

Comment:
Remind me, what's the expected function of `tags`? Is it supposed to select:

1. Topics with any of the listed tags (union)? or:
2. Topics with all of the listed tags (intersection)?

Reply:
2. Intersection :)

Reply:
Could you clarify this in the function description?

But speaking of that, I don't see why this function needs to be specific, as we discussed before. I thought we were going to make it so this could be a generic URL builder. You treat all arrays the same way - you turn them into comma-separated lists. So it seems to me you really just want a function that turns a dictionary of items into a wordpress-style API URL.

And then, as we discussed before, you can move the explicit arguments into the definition of `get_articles`.

Reply:
Ah, alright. Couldnt remember from the ticket. Updated it accordingly.